prog/base.rb

```diff
@@ -25,6 +25,7 @@ def #{name}
 
   def self.semaphore(*names)
     names.map!(&:intern)
+    names << :destroying if names.include?(:destroy)
```

Comment:
this is great. another option without introducing another semaphore would be to mandate that all labels that come after destroy are prefixed with `destroy` (and no label before destroy is prefixed with `destroy`).
```
when_destroy_set? do
     hop_destroy unless @strand.label.start_with? "destroy"
end
```